### PR TITLE
Add prettier and eslint

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
This PR adds prettier as a formatter and eslint as a linter.

We use eslint as a linter and not prettier as eslint is more capable & has more features as far as I know.

Happy to hear / read your suggestions :)

### CODE REVIEW

File: run.js [hunk 1]
```diff
@@ -356,21 +346,18 @@ const run = async () => {
   console.error("Starting screencast...");
   await page.startScreencast({
     format: IMAGE_FORMAT === "jpeg" ? "jpeg" : "png",
-    everyNthFrame: 1
+    everyNthFrame: 1,
   });
   console.error("Recording screencast...");
 
   const reader = awaitifyStream.createReader(process.stdin);
+  // eslint-disable-next-line no-constant-condition
```
Comment (cmfcmf) [inline run.js]: If we drop eslint, this comment can also be removed.

File: package.json [hunk 2]
```diff
@@ -19,6 +23,10 @@
     "uuid": "^8.3.2"
   },
   "devDependencies": {
-    "pkg": "^5.5.1"
+    "eslint": "^8.5.0",
+    "eslint-config-airbnb": "^17.1.0",
+    "eslint-config-prettier": "^8.3.0",
```
Comment (cmfcmf) [inline package.json:25]: The airbnb and prettier configuration aren't actually referenced in the `.eslintrc` file you added.

File: package.json [hunk 3]
```diff
@@ -6,7 +6,11 @@
   },
   "scripts": {
     "build": "pkg --targets node16-win,node16-macos,node16-linux --public --output ./build/magicmouse ./run.js",
-    "postinstall": "patch-package"
+    "postinstall": "patch-package",
+    "pretty": "node_modules/.bin/prettier --write --config .prettierrc \"*.js\"",
+    "am-i-pretty": "node_modules/.bin/prettier --list-different --config .prettierrc \"*.js\"",
+    "lint": "node_modules/.bin/eslint --cache --quiet ./",
+    "lint-fix": "node_modules/.bin/eslint --cache --fix --quiet ."
```
Comment (cmfcmf) [inline package.json]: You don't need to use `node_modules/.bin/FOO`, you can just use `FOO` directly here.

File: getElements.js [hunk 4]
```diff
@@ -138,14 +123,14 @@ const getElements = (fn, ...args) => {
...
       w,
       h,
-      data: includeData ? img2Base64(img) : null
+      data: includeData ? img2Base64(img) : null,
     };
   };
 
   const extractCanvas = async (canvas, includeData = true) => {
-    const rect = img.getBoundingClientRect();
+    const rect = canvas.getBoundingClientRect();
```
Comment (cmfcmf) [inline getElements.js:131]: Good catch!

**Review by cmfcmf — COMMENTED** [review]
Hey @MichaelBuessemeyer,

thanks a lot for this PR! One thing that I would like to discuss is the addition of eslint. I think it is slightly overkill for this project and I am slightly in favor of _not_ adding it (but you can definitely convince me otherwise).

Also, I would prefer if you added just two new scripts to `package.json`: `format` and `lint`, that run prettier (and possibly eslint), where `format` is a combination of the current `pretty` and `lint-fix` scripts and `lint` is a combination of `am-i-pretty` and `lint`.

**Comment by MichaelBuessemeyer** [comment]
> thanks a lot for this PR!

Your welcome :)

> One thing that I would like to discuss is the addition of eslint. I think it is slightly overkill for this project and I am slightly in favor of not adding it (but you can definitely convince me otherwise).

I understand that you think that eslint is overkill. But it has the advantage that it lints the code 😅 . This prevent some errors that might not be obvious at first sight. I only found the error in `element.js` (see above) thanks to eslint.  Therefore using eslint can prevent these errors.

 I am fine if you do not want it as a dependency. But i'll use it in #13 anyway to prevent myself from writing error that the linter can catch easily.

> Also, I would prefer if you added just two new scripts to package.json: format and lint, that run prettier (and possibly eslint), where format is a combination of the current pretty and lint-fix scripts and lint is a combination of am-i-pretty and lint.

Alrighty 🐈

File: package.json (hunk 2 above)
Comment (MichaelBuessemeyer) [reply package.json:25]: Your are right. Looks like I forgot to remove them. I simply tried the same settings from another project, but at least the aribnb dependecy is only for frontend usage and not for backend (nodejs).

File: run.js (hunk 1 above)
Comment (MichaelBuessemeyer) [reply run.js]: I changed the `no-constant-condition` rule in the settings file so that it ignores loops.

Comment (cmfcmf) [reply run.js]: Nice!

**Comment by cmfcmf** [comment]
Alright, thank you! I re-added `eslint-config-prettier` to avoid any potential conflicts between Prettier and ESLint, and added `yarn lint` to the GitHub action.

**Pull request merged**